src/tiled/tilecollisiondock.cpp

```diff
@@ -127,6 +136,51 @@ TileCollisionDock::~TileCollisionDock()
     setTile(nullptr);
 }
 
+/**
+ * @brief TileCollisionDock::autoDetectMask
+ *
+ * Automatically detect the extents of the tile and append a simple rectanglular collision mask.
+ */
+void TileCollisionDock::autoDetectMask()
+{
+    // Iterate over the pixels, looking for empty rows
+    // sourced from: https://stackoverflow.com/a/3722160/2431627
+    QImage image = mTile->image().toImage();
+    int left = image.width(), right = 0, top = image.height(), bottom = 0;
+    for (int y = 0; y < image.height(); ++y) {
+        QRgb *row = (QRgb*)image.scanLine(y);
+        bool rowFilled = false;
+        for (int x = 0; x < image.width(); ++x) {
+            if (qAlpha(row[x])) {
+                rowFilled = true;
+                right = std::max(right, x);
+                if (left > x) {
+                    left = x;
+                    x = right; // shortcut to only search for new right bound from here
+                }
+            }
+        }
+        if (rowFilled) {
+            top = std::min(top, y);
+            bottom = y;
+        }
+    }
```

Comment:
Though actually, after doing some digging in how the `QGraphicsPixmapItem` determines its shape, the following would do the same:
```c++
QRect content = QRegion(mTile->image().createHeuristicMask()).boundingRect();
```
It does involve the allocation of a temporary `QBitmap` and the construction of a `QRegion`, but I think it will not matter much for performance and it keeps this code a lot shorter.